static/js/publisher/form.js

```diff
@@ -381,6 +407,14 @@ function initForm(config, initialState, errors) {
     updateFormState();
   });
 
+  const openPreview = () => {
+    const previewForm = document.getElementById("preview-form");
+    const stateInput = previewForm.elements.state;
+    stateInput.value = JSON.stringify(state);
+  };
+
+  previewButton.addEventListener("click", openPreview);
```

Comment:
Maybe it would be safer to do the handling on `submit` event on preview form rather than click?
The same button does the submit of the form we are modifying, so it kind of relies on the fact that click event is (hopefully) always triggered before submit.
But I guess we could just listen to submit and update the state input there?

Reply:
Yeah, good catch. Originally I was submitting a JS generated form with JS so needed to add the event to the button, but now it's a proper form it makes sense to be on the form "submit" :+1